modules/unsupported/s3-geotiff/src/main/java/org/geotools/s3/S3Connector.java

```diff
@@ -133,10 +137,14 @@ public AmazonS3 getS3Client() {
 
             // aws cli client
         } else if (useAnon) {
-            s3 = new AmazonS3Client(new AnonymousAWSCredentials());
+            s3 =
+                    AmazonS3ClientBuilder.standard()
+                            .withCredentials(
+                                    new AWSStaticCredentialsProvider(new AnonymousAWSCredentials()))
+                            .build();
             s3.setRegion(Region.getRegion(region));
```

Comment:
The `setRegion` call here isn't valid when using the builder pattern, the region needs to be set via the builder.

```
java.lang.RuntimeException: java.lang.UnsupportedOperationException: Client is immutable when created with the builder.
	at org.geotools.s3.geotiff.S3GeoTiffFormat.getReader(S3GeoTiffFormat.java:128) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	at org.geotools.s3.geotiff.S3GeoTiffFormat.getReader(S3GeoTiffFormat.java:43) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	at org.geoserver.catalog.ResourcePool.getGridCoverageReader(ResourcePool.java:1593) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.catalog.ResourcePool.getGridCoverageReader(ResourcePool.java:1509) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.catalog.ResourcePool.getGridCoverageReader(ResourcePool.java:1504) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.catalog.CatalogBuilder.initCoverage(CatalogBuilder.java:858) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.rest.catalog.CoverageController.handleObjectPost(CoverageController.java:326) ~[gs-restconfig-2.22.2.jar:2.22.2]
	at org.geoserver.rest.catalog.CoverageController.coveragePost(CoverageController.java:188) ~[gs-restconfig-2.22.2.jar:2.22.2]
	at jdk.internal.reflect.GeneratedMethodAccessor474.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1043) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681) ~[servlet-api.jar:4.0.FR]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883) ~[spring-webmvc-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) ~[servlet-api.jar:4.0.FR]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-websocket.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:357) ~[catalina.jar:9.0.65]
	at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:176) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:126) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.ThreadLocalsCleanupFilter.doFilter(ThreadLocalsCleanupFilter.java:28) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.SpringDelegatingFilter$Chain.doFilter(SpringDelegatingFilter.java:73) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.ows.HTTPHeadersCollector.doFilter(HTTPHeadersCollector.java:48) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.filters.SpringDelegatingFilter$Chain.doFilter(SpringDelegatingFilter.java:70) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.filters.LoggingFilter.doFilter(LoggingFilter.java:194) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.filters.SpringDelegatingFilter$Chain.doFilter(SpringDelegatingFilter.java:70) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.flow.controller.IpBlacklistFilter.doFilter(IpBlacklistFilter.java:89) ~[gs-control-flow-2.22.2.jar:2.22.2]
	at org.geoserver.filters.SpringDelegatingFilter$Chain.doFilter(SpringDelegatingFilter.java:70) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.monitor.MonitorFilter.doFilter(MonitorFilter.java:144) ~[gs-monitor-core-2.22.2.jar:2.22.2]
	at org.geoserver.filters.SpringDelegatingFilter$Chain.doFilter(SpringDelegatingFilter.java:70) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.filters.SpringDelegatingFilter.doFilter(SpringDelegatingFilter.java:43) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.platform.AdvancedDispatchFilter.doFilter(AdvancedDispatchFilter.java:39) ~[gs-platform-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:71) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:75) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerCompositeFilter.doFilter(GeoServerCompositeFilter.java:92) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:71) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:75) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerCompositeFilter.doFilter(GeoServerCompositeFilter.java:92) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerAnonymousAuthenticationFilter.doFilter(GeoServerAnonymousAuthenticationFilter.java:53) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:71) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilterInternal(BasicAuthenticationFilter.java:215) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:75) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerCompositeFilter.doFilter(GeoServerCompositeFilter.java:92) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerBasicAuthenticationFilter.doFilter(GeoServerBasicAuthenticationFilter.java:81) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:71) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.filter.GeoServerSecurityContextPersistenceFilter$1.doFilter(GeoServerSecurityContextPersistenceFilter.java:52) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerCompositeFilter$NestedFilterChain.doFilter(GeoServerCompositeFilter.java:75) ~[gs-main-2.22.2.jar:2.22.2]
	at org.geoserver.security.filter.GeoServerCompositeFilter.doFilter(GeoServerCompositeFilter.java:92) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178) ~[spring-security-web-5.1.13.RELEASE.jar:5.1.13.RELEASE]
	at org.geoserver.security.GeoServerSecurityFilterChainProxy.doFilter(GeoServerSecurityFilterChainProxy.java:142) ~[gs-main-2.22.2.jar:2.22.2]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.XFrameOptionsFilter.doFilter(XFrameOptionsFilter.java:77) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.GZIPFilter.doFilter(GZIPFilter.java:54) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.SessionDebugFilter.doFilter(SessionDebugFilter.java:49) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.geoserver.filters.FlushSafeFilter.doFilter(FlushSafeFilter.java:42) ~[gs-main-2.22.2.jar:2.22.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.22.RELEASE.jar:5.2.22.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[catalina.jar:9.0.65]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) ~[catalina.jar:9.0.65]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[catalina.jar:9.0.65]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[catalina.jar:9.0.65]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360) ~[catalina.jar:9.0.65]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:890) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-util.jar:9.0.65]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-util.jar:9.0.65]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:9.0.65]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.lang.UnsupportedOperationException: Client is immutable when created with the builder.
	at com.amazonaws.AmazonWebServiceClient.checkMutability(AmazonWebServiceClient.java:808) ~[aws-java-sdk-core-1.11.36.jar:?]
	at com.amazonaws.AmazonWebServiceClient.setRegion(AmazonWebServiceClient.java:334) ~[aws-java-sdk-core-1.11.36.jar:?]
	at com.amazonaws.services.s3.AmazonS3Client.setRegion(AmazonS3Client.java:636) ~[aws-java-sdk-s3-1.11.36.jar:?]
	at org.geotools.s3.S3Connector.getS3Client(S3Connector.java:145) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	at org.geotools.s3.S3ImageInputStreamImpl.getS3Client(S3ImageInputStreamImpl.java:44) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	at org.geotools.s3.S3ImageInputStreamImpl.<init>(S3ImageInputStreamImpl.java:73) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	at org.geotools.s3.geotiff.S3GeoTiffFormat.getReader(S3GeoTiffFormat.java:111) ~[gt-s3-geotiff-28-SNAPSHOT.jar:?]
	... 120 more
```

Reply:
Happy to raise a PR for this once I've had a chance to read the contribution guidelines, just wanted to flag it somewhere.